src/backend/distributed/commands/role.c

```diff
@@ -893,10 +892,31 @@ GenerateGrantRoleStmtsOfRole(Oid roleid)
 		grantRoleStmt->grantor = NULL;
 
 #if PG_VERSION_NUM >= PG_VERSION_16
+
+		/* inherit option is always included */
+		DefElem *inherit_opt;
+		if (membership->inherit_option)
+		{
+			inherit_opt = makeDefElem("inherit", (Node *) makeBoolean(true), -1);
+		}
+		else
+		{
+			inherit_opt = makeDefElem("inherit", (Node *) makeBoolean(false), -1);
```

Comment:
The Codecov warnings that you have here happen because you need to first remove the node before adding it. Right now it is not testing anything.